common/Environments/Helpers/EnvironmentsNotificationHelper.cs

```diff
@@ -214,37 +226,64 @@ private void AddUserToHyperVAdminGroupAndEnableHyperV(Notification notification)
                         ShowErrorWithRebootAfterExecutionMessage(_stringResource.GetLocalized("UserNotAddedToHyperVAdminGroupMessage"));
                         return;
                     case 4:
                         // The user is already in the Hyper-V Admin group and the script successfully enabled the Hyper-Feature.
+                        telemetryEnablementMap.Add(FeatureEnablementKind.HyperVFeature, EnvironmentsTelemetryStatus.Succeeded);
+                        LogEnablementTelemetry(telemetryEnablementMap);
                         _shouldShowHyperVRebootButton = true;
                         ShowRestartNotification();
                         return;
                     case 5:
                         // The user is already in the Hyper-V Admin group and the script failed to enable the Hyper-Feature.
+                        telemetryEnablementMap.Add(FeatureEnablementKind.HyperVFeature, EnvironmentsTelemetryStatus.Failed);
+                        LogEnablementTelemetry(telemetryEnablementMap, "Failed to enable Hyper-V Feature: exit code 5");
                         ShowErrorWithRebootAfterExecutionMessage(_stringResource.GetLocalized("UnableToEnableHyperVFeatureMessage"));
                         return;
                     case 6:
-                        // Display nothing as there is no work to be done
+                        // Display nothing as there is no work to be done, as the feature is enabled and user is in the group.
+                        telemetryEnablementMap.Add(FeatureEnablementKind.HyperVFeature, EnvironmentsTelemetryStatus.NoOperation);
+                        telemetryEnablementMap.Add(FeatureEnablementKind.HyperVAdminGroup, EnvironmentsTelemetryStatus.NoOperation);
+                        LogEnablementTelemetry(telemetryEnablementMap);
                         return;
                 }
+
+                throw new EnvironmentNotificationScriptException($"HyperV Enablement Script failed with exit code: {process.ExitCode}");
             }
             catch (Exception ex)
             {
                 _log.Error(ex, "Script failed, we may not have been able to add user to Hyper-V admin group or enable Hyper-V");
+                telemetryEnablementMap.Add(FeatureEnablementKind.HyperVFeature, EnvironmentsTelemetryStatus.Unknown);
+                telemetryEnablementMap.Add(FeatureEnablementKind.HyperVAdminGroup, EnvironmentsTelemetryStatus.Unknown);
+                LogEnablementTelemetry(telemetryEnablementMap, ex.Message);
             }
 
             ShowErrorWithRebootAfterExecutionMessage(_stringResource.GetLocalized("UnableToAddUserToHyperVAdminAndEnableHyperVMessage"));
         });
+    }
+
+    private void LogEnablementTelemetry(Dictionary<FeatureEnablementKind, EnvironmentsTelemetryStatus> features, string? failureMessage = null)
+    {
+        TelemetryFactory.Get<ITelemetry>().Log(
+            "Environment_FeatureEnablement_Event",
```

Comment:
Considering this is only for HyperV.  The event name should have HyperV in it.  Maybe "Environment_HyperVEnablemenet_Event"

Reply:
For this one, we're going to add enablement for Windows Sandbox in the future.